### PR TITLE
583 Array level random crop by pos / neg label

### LINKED ISSUE #583: Add array level RandCropByPosNegLabel transform
**Is your feature request related to a problem? Please describe.**
Currently, we only have the RandCropByPosNegLabeld for dict data, need a version for array.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #583 .

### Description
This PR implemented the Array version `RandCropByPosNegLabel` transform.
Requested from https://github.com/Project-MONAI/MONAI/issues/452.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/croppad/array.py [hunk 1]
```diff
@@ -319,3 +319,60 @@ def __call__(self, img):
...
+        self.image = image
+        self.image_threshold = image_threshold
+        self.centers = None
+
+    def randomize(self, label, image):
+        self.centers = generate_pos_neg_label_crop_centers(
+            label, self.spatial_size, self.num_samples, self.pos_ratio, image, self.image_threshold, self.R
+        )
+
+    def __call__(self, img, label=None, image=None):
```
Comment (wyli) [inline monai/transforms/croppad/array.py:371]: `img` and `image` could be confusing, could you add some docstring here?

Comment (Nic-Ma) [reply monai/transforms/croppad/array.py]: added doc-string in the latest commit.

**Comment by Nic-Ma** [comment]
> thanks, optionally could you add an example in the docstring?

Hi @wyli ,

Thanks for your review.
I updated the PR based on your comments.
Could you please help review it again?
Thanks.

**Pull request merged**